### DISCUSSION TITLE
Currencies using the same symbol not possible

### DISCUSSION THREAD

**Discussion by Sharparam** [firefly-iii/firefly-iii#6763] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

The Swedish and Norwegian krona/krone currency both use the same currency symbol: "kr". When trying to add both to Firefly III, one or the other will fail depending on which was added first, because the currency symbol is set to be unique.

Presumably this is a problem for other currencies, like several that use the dollar sign but are not the US dollar currency.

### Expected behaviour

I can add the new currency and use it throughout Firefly III.

### Debug information

Debug information generated at 2022-12-29 07:03:58 Europe/Stockholm for Firefly III version **5.7.16**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.16 |
| Firefly III API | 1.5.6 |
| Build | 316, 25-12-2022 07:50:13 CET |
| DB version | 18 (exp. 18) |
| PHP | `8.1.13` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/Stockholm |
| Browser TZ | Europe/Stockholm |
| App environment | local |
| App debug mode | false |
| App cache driver | redis |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | pgsql |
| Trusted proxies (.env) | `**` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2022-12-01 00:00:00 |
| Session end | 2022-12-31 23:59:59 |
| Session first | 2022-01-01 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0 |

### Steps to reproduce

 1. Visit the configuration page for currencies
 2. Add a new currency
 3. Input a symbol that is already used by another currency
 4. Try to add the new currency

### Additional info

Screenshot of issue showing the validation error (which seems to be missing a localization string?):

![image](https://user-images.githubusercontent.com/551094/209910547-144b7f17-df5e-4386-99dc-2d40d73ea510.png)

Given that the symbol seems to be the sole identifier of a currency in many places in the UI, I'm not sure how that would be handled if duplicate symbols become allowed.

**Comment by JC5** [comment]
You can use "NKr" instead?

**Reply by Sharparam** [reply]
It's a possible workaround but it's not the symbol used by the currencies. Though I'm not sure what a proper way would be (and it'd likely involve quite a bit of work).

**Reply by JC5** [reply]
I know that. As another example, there are over 20 currencies that are some variation of "dollar".

**Comment by rtscruggs** [comment]
As I accidentally discovered, Firefly only validates the symbol when the currency is initially created. If you create, say, Swedish krona with the symbol SEK, you can then edit the currency and change the symbol to kr. I don't know if having a duplicate symbol will actually break anything or just look confusing.

I'm not sure how much work it would be to change, but it might be nice to allow duplicate symbols in disabled currencies. (Are the currency symbols displayed anywhere other than than the currency page if currencies are disabled?) This would still leave cases where people wanted to use both NOK and SEK, for example, but it would reduce the errors - and be nicer to new users who are trying to add their currency with a used symbol.

**Reply by JC5** [reply]
> I don't know if having a duplicate symbol will actually break anything or just look confusing.

It's just confusing. But if I allow it, it's also my fault for some reason for allowing people to confuse themselves. This little barrier serves fine for this purpose. I'm not going to spend time adding advanced checks.
